docs/data/data-grid/row-grouping/row-grouping.md

```diff
@@ -377,6 +377,77 @@ Row selection propagation has some limitations:
 
 :::
 
+## Reorder row grouping rows
+
+Row reordering allows the users to drag and drop the rows to update their order.
+To enable this feature with the row grouping, pass the `rowReordering` prop on the Data Grid Premium component:
+
+```tsx
+<DataGridPremium rowGroupingModel={['category']} rowReordering />
+```
+
+{{"demo": "RowGroupingReordering.js", "bg": "inline", "defaultCodeOpen": false}}
```

Comment:
I noticed that if I move the row to a different group, and then stop grouping by company, the row data is reverted to the original.
Is this expected?

Reply:
>  Is this expected?

No, it was a bug. The row update was incomplete. Fixed: https://github.com/mui/mui-x/pull/18251/commits/fcf5d17c1bbbe7babc9ef285a53c954c56f2e90f